src/lib/services/email-service.ts

```diff
@@ -138,6 +140,86 @@ export class EmailService {
         });
     }
 
+    async sendScheduledChangeConflictEmail(
+        recipient: string,
+        conflictScope: 'flag' | 'strategy',
+        changeRequests: {
+            id: number;
+            scheduledAt: string;
+            title?: string;
+        }[],
+        strategyIdOrFlagName: string,
```

Comment:
Maybe we could use types to make this clearer? This looks like two different conflict types. What happens if you send `conflictScope: 'strategy', strategyIdOrFlagName: 'my-feature-flag'`? 

It'll be treated as a strategy, even if you send a flag name, right?

We could do something like this

```ts
type StrategyConflict = {
  conflictScope: 'strategy',
  strategyId: string;
}

type FlagConflict = {
  conflictScope: 'flag',
  flagName: string;
}
```

and then change the signature to:

```ts
        recipient: string,
        changeRequests: {
            id: number;
            scheduledAt: string;
            title?: string;
        }[],
        conflictData: StrategyConflict | FlagConflict
```

Or we could use a union type for the whole thing.

It might not be necessary, but it just struck me that these arguments make it easy to get into states that are not what you mean.

Reply:
I revised it a bit so that we always get a flag name and project